### PR TITLE
Add hint to Falsy Bouncer challenge description

### LINKED ISSUE #12433: Falsy Bouncer kluge coding
Challenge [Falsy Bouncer](https://www.freecodecamp.com/challenges/falsy-bouncer#?solution=%0Afunction%20bouncer(arr)%20%7B%0A%20%20%2F%2F%20Don%27t%20show%20a%20false%20ID%20to%20this%20bouncer.%0A%20function%20notFalse%20(item)%20%7B%0A%20%20%20var%20x%20%3D%20typeof(item)%3B%0A%2F%2F%20%20%20console.log(%22%20item%20%22%20%2B%20item%20%2B%20%22%20x-typeof%20%22%20%2B%20x)%3B%0A%2F%2F%20%20%20if%20(x%20%3D%3D%3D%20%22number%22)%0A%2F%2F%20%20%20%20%20if%20(!item)%0A%2F%2F%20%20%20%20%20%20%20item%20%3D0%3B%20%20%2F%2F%20force%20NaN%20to%20be%200%0A%20%20%20%0A%20%20%20%0A%20%20%20if%20(item%20%3D%3D%3D%20false)%20%7B%0A%20%20%20%20%20return%20false%3B%0A%20%20%20%7D%20else%20if%20(item%20%3D%3D%3D%20%22%22)%20%7B%0A%20%20%20%20%20return%20false%3B%0A%20%20%20%7D%20else%20if%20(item%20%3D%3D%3D%20null)%20%7B%0A%20%20%20%20%20return%20false%3B%0A%20%20%20%7D%20else%20if%20(item%20%3D%3D%3D%200)%20%7B%0A%20%20%20%20%20return%20false%3B%0A%20%20%20%7D%20else%20if%20(item%20%3D%3D%3D%20undefined)%20%7B%0A%20%20%20%20%20return%20false%3B%0A%20%20%20%7D%20else%20if%20(x%20%3D%3D%3D%20%22number%22%20%26%26%20isNaN(item))%20%7B%20%20%2F%2F%20%22a%22%20would%20fail%20this%20test%20without%20%3D%3D%3D%20number%0A%20%20%20%20%20return%20false%3B%0A%20%20%20%7D%20else%20%7B%0A%20%20%20%20%20return%20true%3B%0A%20%20%20%7D%20%20%20%0A%20%7D%0A%20%20%20%20%20%20%0A%20%20%20%20%20%20%0Avar%20newArr%20%3D%20arr.filter(notFalse)%3B%20%20%20%20%20%20%0A%20%20%20%20%0A%20%20%0A%20%20return%20newArr%3B%0A%7D%0A%0Abouncer(%5B7%2C%20%22ate%22%2C%20null%2C%20false%2C%209%5D)%3B%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10.6; rv:48.0) Gecko/20100101 Firefox/48.0</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```javascript

function bouncer(arr) {
  // Don't show a false ID to this bouncer.
 function notFalse (item) {
   var x = typeof(item);
//   console.log(" item " + item + " x-typeof " + x);
//   if (x === "number")
//     if (!item)
//       item =0;  // force NaN to be 0
   
   
   if (item === false) {
     return false;
   } else if (item === "") {
     return false;
   } else if (item === null) {
     return false;
   } else if (item === 0) {
     return false;
   } else if (item === undefined) {
     return false;
   } else if (x === "number" && isNaN(item)) {  // "a" would fail this test without === number
     return false;
   } else {
     return true;
   }   
 }
      
      
var newArr = arr.filter(notFalse);      
    
  
  return newArr;
}

bouncer([7, "ate", null, false, 9]);

```
*************************************
My comments:
This was a challenging exercise.  I have read the other comments in this forum and understand that this is a borderline bug/quirk of JS coding that is unlikely to change.
The "falsy" concept is not monolithic when it comes to testing for all the possible falsy values.  NaN is the exception and should only be applied when the typeof (item) === "number" whereas the other falsy concepts can be tested regardless of object type.  
I wonder if the typeof () function should be added to the the helpful links on this challenge?  
I also wonder if it would not be better to test for the falsy concepts AFTER determining the object type, for all falsy results?
Main issue here is that the function isNaN(x) will return true for any string/character values tested (probably boolean as well but my code tests for the boolean falsy prior to the NaN test).

I know I should likely be using the chat/help function for this and while new at freecodecamp I expect to be up and running on github shortly.

### PR DESCRIPTION
**Pull request by codeman869**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #12433

#### Description

Updates the Falsy Bouncer challenge description to include a hint
to help campers understand how they should approach the challenge
without revealing too much information or locking the camper into
a particular solution.

### CODE REVIEW

File: seed/challenges/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting.json [hunk 1]
```diff
@@ -685,6 +685,7 @@
       "description": [
         "Remove all falsy values from an array.",
         "Falsy values in JavaScript are <code>false</code>, <code>null</code>, <code>0</code>, <code>\"\"</code>, <code>undefined</code>, and <code>NaN</code>.",
+        "Hint: Try converting each value to Boolean.",
```
Comment (systimotic) [inline seed/challenges/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting.json]: To **a** Boolean

**Review by systimotic — CHANGES_REQUESTED** [review]
Thanks for the PR @codeman869!

**Comment by codeman869** [comment]
@systimotic - I've amended my commit!

**Review by Greenheart — APPROVED** [review]
@codeman869 Nice work! :smile:

**Review by systimotic — APPROVED** [review]
# LGTM 👍 🎉 
Thank you for contributing again @codeman869! Also a thank you for helping figure out what to put here!

**Pull request merged**